### PR TITLE
Core: fix holdReady and move readywait to an iframe test

### PR DESCRIPTION
**Pull request by timmywil**
Close gh-3331
Fixes gh-3573

### Summary ###
I've edited @stevemao's PR, made the recommended changes, and fixed the issue with `jQuery.holdReady`. I confirmed that the test failed before moving `holdReady` back to its proper place.

### Checklist ###

* [x] All authors have signed the CLA at https://contribute.jquery.com/CLA/
  - Steve signed the jQuery CLA, which is enough to land this PR.
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes

### CODE REVIEW

**Comment by timmywil** [comment]
Will land today. Get your reviews in! Thanks.

File: test/unit/readywait.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var start = new Date();
```
Comment (mgol) [inline test/unit/readywait.js]: What's the purpose of this file? It runs the same test as one in test/unit/ready and it's not used anywhere, this looks like dead code.

Comment (timmywil) [inline test/unit/readywait.js]: Indeed. I thought I deleted it.

Comment (timmywil) [inline test/unit/readywait.js]: Deleted.

**Pull request closed without merging**